src/tests/fixing.rs

```diff
@@ -0,0 +1,159 @@
+#![cfg(test)]
+
+use std::sync::Arc;
+
+use tree_sitter::Node;
+use tree_sitter_grep::SupportedLanguage;
+
+use crate::{
+    context::QueryMatchContext,
+    rule::{FileRunInfo, Rule, RuleInstance, RuleInstancePerFile, RuleListenerQuery, RuleMeta},
+    run_fixing_for_slice, Config, ConfigBuilder, ViolationBuilder,
+};
+
+#[test]
+fn test_single_fix() {
+    let mut file_contents = r#"
+        fn foo() {}
+    "#
+    .to_owned()
+    .into_bytes();
+    run_fixing_for_slice(
+        &mut file_contents,
+        "tmp.rs",
+        ConfigBuilder::default()
+            .rules([create_identifier_replacing_rule("foo", "bar")])
+            .fix(true)
+            .build()
+            .unwrap(),
+    );
+    assert_eq!(
+        std::str::from_utf8(&file_contents).unwrap().trim(),
+        r#"
+            fn bar() {}
+        "#
+        .trim()
+    );
+}
+
+struct IdentifierReplacingRule {
+    name: String,
```

Comment:
So this is demonstrating there being "state" (in this case basically "static" state) in a rule

There are probably ways to avoid "passing this state from `Rule` -> `RuleInstance` -> `RuleInstancePerFile` via `.clone()`'ing" but whatever (for the moment)